src/library/interface_stereo.cpp

```diff
@@ -6,38 +6,131 @@ namespace orb_slam_2_interface {
       new message_filters::Subscriber<sensor_msgs::Image>(
-          nh_, "camera/right/image_raw", 1));
+          nh_, "cam1/image_raw", 1));
   // Creating a synchronizer
   sync_ = std::shared_ptr<message_filters::Synchronizer<sync_pol>>(
       new message_filters::Synchronizer<sync_pol>(sync_pol(10), *left_sub_,
                                                   *right_sub_));
   // Registering the synchronized image callback
   sync_->registerCallback(
       boost::bind(&OrbSlam2InterfaceStereo::stereoImageCallback, this, _1, _2));
+
+}
+
+bool OrbSlam2InterfaceStereo::getBodyTransform()
+{
+
+  cv::FileStorage fsSettings(settings_file_path_, cv::FileStorage::READ);
+  if(!fsSettings.isOpened())
+  {
+      ROS_ERROR("ERROR: Wrong path to settings");
+      return false;
+  }
+
+  cv::Mat T_C0_B;
+  Transformation T_C_B;
+
+  fsSettings["T_c0_fcuimu"] >> T_C0_B;
+  convertOrbSlamPoseToKindr(T_C0_B, &T_C_B);
+  T_B_C_ = T_C_B.inverse();
+
+  return true;
+}
+
+bool OrbSlam2InterfaceStereo::stereoRectification()
+{
+    // Load settings related to stereo calibration
+    cv::FileStorage fsSettings(settings_file_path_, cv::FileStorage::READ);
+    if(!fsSettings.isOpened())
+    {
+        ROS_ERROR("ERROR: Wrong path to settings");
+        return false;
+    }
+    
+    cv::Mat K_l, K_r, P_l, P_r, R_l, R_r, D_l, D_r, T_C0_1C, Q_;
+
+    fsSettings["LEFT.K"] >> K_l;
+    fsSettings["RIGHT.K"] >> K_r;
+
+    fsSettings["LEFT.P"] >> P_l;
+    fsSettings["RIGHT.P"] >> P_r;
+
+    fsSettings["LEFT.R"] >> R_l;
+    fsSettings["RIGHT.R"] >> R_r;
+
+    fsSettings["LEFT.D"] >> D_l;
+    fsSettings["RIGHT.D"] >> D_r;
+
+    fsSettings["T_c0_c1"] >> T_C0_1C;
+
+    int rows_l = fsSettings["LEFT.height"];
+    int cols_l = fsSettings["LEFT.width"];
+    int rows_r = fsSettings["RIGHT.height"];
+    int cols_r = fsSettings["RIGHT.width"];
+
+    if(K_l.empty() || K_r.empty() || P_l.empty() || P_r.empty() || R_l.empty() || R_r.empty() || D_l.empty() || D_r.empty() ||
+            rows_l==0 || rows_r==0 || cols_l==0 || cols_r==0)
+    {
+        ROS_ERROR("ERROR: Calibration parameters to rectify stereo are missing!");
+        return false;
+    }
+
+    cv::stereoRectify(K_l,D_l,K_r,D_r,cv::Size(cols_l,rows_l),T_C0_1C.rowRange(0,3).colRange(0,3),T_C0_1C.col(3).rowRange(0,3),
+    R_l,R_r,P_l,P_r,Q_);
+
+    cv::initUndistortRectifyMap(K_l,D_l,R_l,P_l.rowRange(0,3).colRange(0,3),cv::Size(cols_l,rows_l),CV_32F,M1l_,M2l_);
+    cv::initUndistortRectifyMap(K_r,D_r,R_r,P_r.rowRange(0,3).colRange(0,3),cv::Size(cols_r,rows_r),CV_32F,M1r_,M2r_);
+
+    //Uncomment if you want to see the rectified instrinsics ex. to put in yaml file
+
+    /*
+    cout << "LEFT.R" << R_l << endl;
+    cout << "RIGHT.R" << R_r << endl;
+    cout << "LEFT.P" << P_l << endl;
+    cout << "RIGHT.P" << P_r << endl;
+    cout << "Baseline" << -1*P_r.at<float>(0,3) << endl; // baseline
+    */
+
+    stereo_rectified_ = true;
+
+    return true;
```

Comment:
We have a default node for stereo rectification here:

https://github.com/ethz-asl/image_undistort

if you find that you need more functionality. That said, this seems fairly light weight so it's fine to keep it in, with the obvious exception that params should come from the param server